### ISSUE TITLE
Vorläufige Korrektur

### ISSUE THREAD

**Issue by Kegny** [ProPra16/programmierpraktikum-abschlussprojekt-die-bogo-bande#1]
Eure Abgabe ist in fast Ordnung; Ihr müsst noch Gradle fixen.
Ansonsten würde ich gerne noch etwas mehr von IWummi sehen (Dominique hat bisher gar keinen Code),
das bisherige ist etwas zu wenig. Du könntest dich z.B. an die Bugs setzten und
im Ilias auf die von Jens gemachten Vorschläge eingehen.

Zu beheben:
- [x] Gradle; Das builded bei euch einfach daher, da Ihr nur minimalstes einfügt.
  Das gebaute soll eigtl. in irgendeiner Form euer Programm und ausführbar sein

Kleinere Mängel:
- [x] Man kann den Zyklus nicht verlassen; Es gibt keinen Zurückbutton ins Menü
- [x] Euer Speichern hat Probleme und logt die ganze Zeit Fehlermeldungen in die Konsole
  (Bitte jetzt nicht einfach die System.out. entfernen)
- [x] Bei euch schließt das Programm nicht vollständig. Es verbleibt noch immer mindestens 1
  Prozess

Ansonsten ist das eine solide Abgabe mit sehr gelungenem Tracking und die Babysteps funktionieren.

**Comment by Kegny** [comment]
Hej, ich sehe, dass wieder gearbeitet wird, wie sieht es bei euch so vom Fortschritt her aus? Kommt ihr zurecht?

**Comment by IWummi** [comment]
Wir beheben gerade die kleinen Mängel, läuft soweit gut.
Wäre nett, wenn du nochmal genauer sagen kannst, was du mit den Punkt unter "zu Beheben" meinst, ich habe dass nicht so richtig verstanden.
Und unter dem Punkt, dass unser Programm Speicherprobleme hat: Meinst du die Fehler die auftreten, wenn die TXT-Datei nicht existiert??Ansonsten hatte ich nämlich keine Fehler bekommen

**Comment by Kegny** [comment]
Nein, die Speicherprobleme dürften nicht, davon kommen, dass eine Datei nicht gefunden wird. Das ist ein Thread, der entweder von euch oder indirekt gestartet, aber nicht geschlossen wird. Von meinen eigenen Erfahrungen her würde ich jetzt einfach mal vermuten, dass bei eurem Sound ein .open() oder so aufgerufen werden muss, aber dann halt auch ein .close() zum schließen, was ihr vergessen habt (REINE VERMUTUNGEN).
Außerdem ist ein nicht-schließender Thread ja auch in der Hinsicht kein Fehler, der z.B. Exceptions wirft oder einen Absturz verursacht. Es ist ein Prozess, der im Hintergrund läuft und Leistung zieht, ohne Sinnvoll zu sein (Das eigtl Programm ist ja durchgelaufen)

Was das einzige, was für das ausreichen des Projektes zu beheben ist, betrifft:
Teil der Aufgabe ist es mit GRADLE zu bauen(Vergleichbar mit einem make-File oder einem Kompilier-How-To), welches dann als zusätzliche Test-Maßnahme von Travis auf einem leeren System geklont und mit dem gradle-Skript gebaut wird.

Immoment erzeugt Ihr ein .tar.gz, welches lediglich die Ressourcen beinhaltet, aber nicht z.B. die Kompilierten Klassen. Das ergebnis von "gradle build" oder vergleichbarem, sollte aber euer Programm sein, wie Ihr es z.B. immoment über die IDE startet, oder z.B. Minecraft. Das ist ja auch eine .jar (wobei auf Windows idR, über einen .exe-Launcher gestartet).
Habe ich das so jetzt besser erklären können?

Vielleicht ist noch https://docs.gradle.org/current/userguide/java_plugin.html#N152C8 hilfreich.

**Comment by Kegny** [comment]
Noch einmal der Hinweis:
Was Projekt ist erst in Ordnung, wenn Ihr Gradle verwendet (Teil der Aufgabenstellung)

**Comment by Kegny** [comment]
@IWummi  Passiert da noch was? Seit den neueren Commits ist euer Timer kaputt und
Tracking loggt scheinbar nichts mehr. Man kann außerdem das initiale Testschreiben nicht mehr verlassen(war vorher auch schon buggy).

Dein Fix wegen des Nicht-Terminierens scheint auch nicht zu greifen..
An Gradle scheint Ihr euch bisher auch nicht gesetzt zu haben.

Vom bisherigem Stand `42d37d509bc6a2635022ad158ca00d5f3e7fa681` würde nur @svgas100 bestehen.

**Comment by Kegny** [comment]
@IWummi brauchst dir keine sorgen mehr zu machen. Das reicht jetzt :)

**Comment by IWummi** [comment]
Alles klar :-)
